lms/djangoapps/course_structure_api/v0/views.py

```
@@ -0,0 +1,236 @@
+        try:
+            course_id = self.kwargs.get('course_id')
+            course_key = CourseKey.from_string(course_id)
+            course = courses.get_course(course_key)
+
+            self.check_course_permissions(self.request, course)
+
+            return course
+        except ValueError:
+            raise Http404
+
+
+class CourseList(CourseViewMixin, ListAPIView):
+    """
+    **Use Case**
+
+        CourseList returns paginated list of courses in the edX Platform. The list can be
+        filtered by course_id
+
+    **Example Request**
+
+          GET /
+          GET /?course_id={course_id1},{course_id2}
+
+    **Response Values**
+
+        * category: The type of content. In this case, the value is always "course".
+
+        * name: The name of the course.
+
+        * uri: The URI to use to get details of the course.
+
+        * course: The course number.
+
+        * due:  The due date. For courses, the value is always null.
+
+        * org: The organization specified for the course.
+
+        * id: The unique identifier for the course.
+    """
+    paginate_by = 10
+    paginate_by_param = 'page_size'
+    pagination_serializer_class = PaginationSerializer
+    serializer_class = serializers.CourseSerializer
+
+    def get_queryset(self):
+        course_ids = self.request.QUERY_PARAMS.get('course_id', None)
+
+        course_descriptors = []
+        if course_ids:
+            course_ids = course_ids.split(',')
+            for course_id in course_ids:
+                course_key = CourseKey.from_string(course_id)
+                course_descriptor = courses.get_course(course_key)
+                course_descriptors.append(course_descriptor)
+        else:
+            course_descriptors = modulestore().get_courses()
+
+        results = [course for course in course_descriptors if self.user_can_access_course(self.request.user, course)]
+
+        if not settings.DEBUG and len(results) == 0:
+            raise PermissionDenied
```

Why raise an error here instead of just returning an empty list?